apps/frontend/core-dapp/src/features/withdraw/WithdrawStore.ts

```diff
@@ -264,21 +268,30 @@ export class WithdrawStore {
     if (withdrawalAmountInWstEthBN === undefined) return
 
     try {
-      const withdrawalAmountInEthBN = await this.root.balancerStore.quoteWstEthAmountInEth(
+      const withdrawalAmountInEth = await this.root.balancerStore.quoteWstEthAmountInEth(
         withdrawalAmountInWstEthBN
       )
 
-      if (withdrawalAmountInEthBN !== undefined) {
+      if (withdrawalAmountInEth !== undefined) {
         runInAction(() => {
-          this.withdrawalMarketValueInEth = { status: 'queried', value: withdrawalAmountInEthBN }
+          if (
+            this.root.advancedSettingsStore.isPriceImpactTooHigh(withdrawalAmountInEth.priceImpact)
+          ) {
+            this.withdrawalMarketValueInEth = { status: 'price-impact-too-high' }
+          } else {
+            this.withdrawalMarketValueInEth = {
+              status: 'queried',
+              value: withdrawalAmountInEth.value,
+            }
+          }
```

Comment:
I think we can undo the changes here, and just add a `highImpact` prop to `status: 'queried'`

Because technically you will still have an estimated received amount, it's just that we disable the button to not let you perform a bad trade.

```suggestion
          this.withdrawalMarketValueInEth = {
              status: 'queried',
              value: withdrawalAmountInEth.value,
              highImpact: this.root.advancedSettingsStore.isPriceImpactTooHigh(withdrawalAmountInEth.priceImpact)
            }
```

alternative would be to still pass value to when status is 'price-impact-too-high'

Reply:
I'm a bit reluctant to keeping it on `queried`, because we rely on `queried` to know whether a trade is valid (like [here](https://github.com/prepo-io/prepo-monorepo/blob/617f4450f83ca8ea8b7821e3a86d7e118699df30/apps/frontend/core-dapp/src/features/withdraw/WithdrawStore.ts#L142) or [here](https://github.com/prepo-io/prepo-monorepo/blob/617f4450f83ca8ea8b7821e3a86d7e118699df30/apps/frontend/core-dapp/src/features/withdraw/WithdrawStore.ts#L154))

So I'll do the second suggestion instead and keep the value on `price-impact-too-high`. I'll also rename `queried` to `trade-valid` for more clarity.